argoverse/evaluation/detection/utils.py

```diff
@@ -99,15 +100,28 @@ class DetectionCfg(NamedTuple):
     map_root: _PathLike = Path(__file__).parent.parent.parent.parent / "map_files"  # argoverse-api/map_files
 
 
-def accumulate(
-    dt_root_fpath: Path, gt_fpath: Path, cfg: DetectionCfg, avm: Optional[ArgoverseMap]
-) -> Tuple[DefaultDict[str, np.ndarray], DefaultDict[str, int]]:
-    """Accumulate the true/false positives (boolean flags) and true positive errors for each class.
+@dataclass
+class AccumulateJob:
+    """Dataclass args for running `accumulate`.
 
     Args:
         dt_root_fpath: Detections root folder file path.
         gt_fpath: Ground truth file path.
         cfg: Detection configuration.
+        avm: Argoverse map object.
+    """
+
+    dt_root_fpath: Path
+    gt_fpath: Path
+    cfg: DetectionCfg
+    avm: Optional[ArgoverseMap]
+
+
+def accumulate(job: AccumulateJob) -> Tuple[DefaultDict[str, np.ndarray], DefaultDict[str, int]]:
```

Comment:
What do you think the best way for us to pass this into a mapping function would be (e.g., `ProcessMap`?

Reply:
Discussed offline on slack. I think this is fine!